samples/kotlin_compose/src/main/java/io/customer/android/sample/kotlin_compose/data/repositories/PreferenceRepository.kt

```diff
@@ -43,23 +45,40 @@ class PreferenceRepositoryImp(private val dataStore: DataStore<Preferences>) :
                 siteId = preferences[SITE_ID] ?: BuildConfig.SITE_ID,
                 apiKey = preferences[API_KEY] ?: BuildConfig.API_KEY
             ).apply {
-                trackUrl = preferences[TRACK_API_URL_KEY]
-                preferences[BACKGROUND_QUEUE_SECONDS_DELAY]?.let {
-                    backgroundQueueSecondsDelay = it
-                }
-                preferences[BACKGROUND_QUEUE_MIN_NUM_TASKS]?.let {
-                    backgroundQueueMinNumTasks = it
-                }
-                preferences[TRACK_SCREEN]?.let {
-                    trackScreen = it
-                }
-                preferences[TRACK_DEVICE_ATTRIBUTES]?.let {
-                    trackDeviceAttributes = it
-                }
-                preferences[DEBUG_MODE]?.let {
-                    debugMode = it
-                }
+                trackUrl =
+                    preferences[TRACK_API_URL_KEY]?.ifEmpty { "https://track-sdk.customer.io/" }
+                        ?: "https://track-sdk.customer.io/"
+
+                backgroundQueueSecondsDelay = preferences[BACKGROUND_QUEUE_SECONDS_DELAY]
+                    ?: CustomerIOConfig.Companion.AnalyticsConstants.BACKGROUND_QUEUE_SECONDS_DELAY
+
+                backgroundQueueMinNumTasks = preferences[BACKGROUND_QUEUE_MIN_NUM_TASKS]
+                    ?: CustomerIOConfig.Companion.AnalyticsConstants.BACKGROUND_QUEUE_MIN_NUMBER_OF_TASKS
+
+                trackScreen = preferences[TRACK_SCREEN] ?: true
+
+                trackDeviceAttributes = preferences[TRACK_DEVICE_ATTRIBUTES] ?: true
+
+                debugMode = preferences[DEBUG_MODE] ?: true
             }
         }
     }
+
+    override suspend fun restoreDefaults(): Configuration {
+        val configuration = Configuration(
+            siteId = BuildConfig.SITE_ID,
+            apiKey = BuildConfig.API_KEY
+        ).apply {
+            trackUrl = "https://track-sdk.customer.io/"
+            backgroundQueueSecondsDelay =
+                CustomerIOConfig.Companion.AnalyticsConstants.BACKGROUND_QUEUE_SECONDS_DELAY
+            backgroundQueueMinNumTasks =
+                CustomerIOConfig.Companion.AnalyticsConstants.BACKGROUND_QUEUE_MIN_NUMBER_OF_TASKS
+            trackScreen = true
+            trackDeviceAttributes = true
+            debugMode = true
+        }
```

Comment:
Similar conversation had [in java sample app](https://github.com/customerio/customerio-android/pull/217#discussion_r1229881347) about strategy of populating default SDK config objects.

Reply:
I think Rehan mentioned it there as well, it's the requirements for the sample app. For example for all the config to be true which by default of the SDK aren't. So we need to set default values

Reply:
This was resolved in Slack. Hard-coded values are good to use 👍